### PR TITLE
Ability to detect SQL queries without bind parameters #9

### PR DESCRIPTION
**Pull request by debbeca**

### CODE REVIEW

File: doc/database_diagnostics.md [hunk 1]
```diff
@@ -352,3 +352,41 @@ Hibernate: select team0_.id as id1_1_0_, team0_.name as name2_1_0_ from team tea
...
+           -- void setSqlWithoutBindParamDetected(boolean)
+```
+
+:mag_right: Log example
+
+```
+Hibernate: SELECT * FROM Player p where p.lastName like '%ronaldo%'
+2022-12-14 13:59:19.776  WARN 13632 --- [qtp640585435-24] s.QuickPerfHttpCallHttpCallWarningLogger : 
+GET 200 http://localhost:8080/player/unbounded-params?name=ronaldo
+	* [WARNING] Some of your SQL queries does not use bound parameters.
```
Comment (jeanbisutti) [inline doc/database_diagnostics.md:391]:
```suggestion
	* [WARNING] Some of your SQL queries don't  use bind parameters.
```

File: spring-boot-2/src/main/java/org/quickperf/web/spring/QuickPerfAfterRequestServletFilter.java [hunk 2]
```diff
@@ -259,6 +262,13 @@ private void quickPerfProcessing(HttpServletRequest httpServletRequest, HttpServ
 
         }
 
+        if(databaseConfig.isSqlWithoutBindParamDetected()) {
+            BooleanMeasure hasBoundParams = AllParametersAreBoundExtractor.INSTANCE.extractPerfMeasureFrom(sqlExecutions);
+            if ( !hasBoundParams.getValue() ) {
+                warnReport.append(lineSeparator() + "\t* [WARNING] Some of your SQL queries does not use bound parameters." );
```
Comment (jeanbisutti) [inline spring-boot-2/src/main/java/org/quickperf/web/spring/QuickPerfAfterRequestServletFilter.java:268]:
```suggestion
                warnReport.append(lineSeparator() + "\t* [WARNING] Some of your SQL queries don't  use bind parameters." );
```

File: spring-boot2-test/src/test/java/messages/warning/SqlWithoutBindParametersWarningRaisedTest.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    should_warn_when_query_does_not_have_bind_parameters() {
+
+        databaseConfig.setSqlWithoutBindParamDetected(true);
+
+        performGetOnUrlReturningJson(FIND_PLAYER_WITH_UNBOUNDED_PARAMS+"?name=Ronaldo");
+
+        TestableQuickPerfMessage quickPerfWarnings = warningInterceptor.getWarnings();
+
+        assertThat(quickPerfWarnings)
+                .hasAMessageContaining("[WARNING] Some of your SQL queries does not use bound parameters");
```
Comment (jeanbisutti) [inline spring-boot2-test/src/test/java/messages/warning/SqlWithoutBindParametersWarningRaisedTest.java:39]:
```suggestion
                .hasAMessageContaining("[WARNING] Some of your SQL queries don't  use bind parameters.");
```

**Review by jeanbisutti — COMMENTED** [review]
Thank you very much for this work @debbeca!

I have done a few suggestions. What do you think?

**Comment by debbeca** [comment]
> Thank you very much for this work @debbeca!
> 
> I have done a few suggestions. What do you think?

all fixed :) thanks for the code review

**Comment by jeanbisutti** [comment]
Thank you @debbeca!

**Pull request merged**